src/cipher.rs

```diff
@@ -509,102 +511,296 @@ impl rustls_root_cert_store {
     }
 }
 
+/// A builder for a `rustls_allow_any_authenticated_client_verifier`. This builder object can be
+/// used to configure certificate revocation lists, and then turned into a
+/// `rustls_allow_any_authenticated_client_verifier` once ready.
+pub struct rustls_allow_any_authenticated_client_builder {
+    _private: [u8; 0],
+}
+
+impl CastPtr for rustls_allow_any_authenticated_client_builder {
+    // XXX: contained value is consumed even on error, so this can contain None. but the caller
```

Comment:
In my personal lexicon, XXX means "resolve before merge." I think this comment can be deleted now that this is an intentional part of the API? Or turned into a "note"?

Reply:
Switched to a `NOTE: ...` - it might be slightly redundant with the API but I found the comment helpful as someone relatively new to the C/Rust boundary and expectations w.r.t `free`.